### ISSUE TITLE
Add pi-temps.py script for Raspberry Pi GPIO monitoring

### ISSUE THREAD

**Issue by geerlingguy** [geerlingguy/temperature-monitor#12]
See the [DS18B20 Guide for Raspberry Pi](https://learn.adafruit.com/adafruits-raspberry-pi-lesson-11-ds18b20-temperature-sensing/overview) on Adafruit. It should be even simpler to set up than Arduino-based serial logging, since it's more internal to Raspbian.

Basically, modify the `arduino-temps.py` script to work with the GPIO pins on a Raspberry Pi.

It would also be pretty nice if I could get multi-sensor configuration to be pretty easy (e.g. just build a map of which sensor on which GPIO pin goes to which `sensor_id`...); that way I could grab a spare/cheap A+ with an Edimax 802.11n chip, and toss it pretty much anywhere in the house with a couple sensors for different rooms, run through walls/ceilings!

**Comment by geerlingguy** [comment]
To get data from the GPIO pin 4 (using [this pinout diagram](http://pi.gadgetoid.com/pinout)), it seems the process to follow is:

```
sudo modprobe w1-gpio && sudo modprobe w1-therm
ls /sys/bus/w1/devices
# Change 'xxxx' below to the device ID.
cd /sys/bus/w1/devices/28-xxxx
cat w1_slave
```

...however, I'm not getting any devices when I list the `devices` directory. I tried GPIO 1 and GPIO 4 both, also verified 3.3V across the red-to-black, and 3.0V across red to data from the DS18B20 waterproof sensor. Next step is to see if maybe this sensor's a dud. I'll test it in an Uno that I know is working correctly.

The command `ls -Al /sys/class/gpio/` lists a `gpio4` symlink:

```
$ ls -Al /sys/class/gpio/
total 0
-rwxrwx--- 1 root gpio 4096 Feb  8 17:01 export
lrwxrwxrwx 1 root gpio    0 Feb  8 16:43 gpio4 -> ../../devices/soc/20200000.gpio/gpio/gpio4
lrwxrwxrwx 1 root gpio    0 Dec 31  1969 gpiochip0 -> ../../devices/soc/20200000.gpio/gpio/gpiochip0
-rwxrwx--- 1 root gpio 4096 Dec 31  1969 unexport
```

So who knows? I'll keep digging.

**Comment by geerlingguy** [comment]
I also tried adding `untested` to the deb sources list in `/etc/apt/sources.list.d/raspi.list`, then `apt-get update && apt-get upgrade`, but that showed no updates available. (As seen in [this guide](https://www.cl.cam.ac.uk/projects/raspberrypi/tutorials/temperature/)).

Hmm.

**Comment by geerlingguy** [comment]
I've also installed WiringPi and run the [WiringPi pintest](http://wiringpi.com/the-gpio-utility/pin-test/), and here's the output:

```
~/wiringPi/gpio $ ./pintest
          The main 8 GPIO pins  0: 7:  OK
                The 5 SPI pins 10:14:  OK
               The serial pins 15:16:  OK
                  The I2C pins  8: 9:  OK
```

**Comment by geerlingguy** [comment]
I also added the lines below to `/etc/modules`, so they load on boot:

```
w1-gpio
w1-therm
```

**Comment by geerlingguy** [comment]
...and it turns out I needed to add the line `dtoverlay=w1-gpio` to my `/boot/config.txt`, due to a recent kernel update for the Raspberry Pi ([see forum thread](http://www.raspberrypi.org/forums/viewtopic.php?f=37&t=98407)). Teaches me for not having configured the DS18B20 using the Pi's GPIO earlier!

I've probably spent an hour quadruple-checking all connections, testing my wires, testing the resistor, using two different sets of jumpers and a spare DS18B20—even going so far as to test the tester with another probe! Ugh, it's always software :P

I also needed to fix my Pi's SD card as I borked it in between these steps, resulting in a solid green ACT light on the Pi :(: http://www.midwesternmac.com/blogs/jeff-geerling/mounting-raspberry-pis-ext4-sd
